File: OIS/ois_findperm/findperm.pas
{ NOTE: it is recommended to use this even if you don't understand the following code }

var
    N, A     : longint;

begin
{
    uncomment the following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    readln(N);

    { SAMPLE INTERACTION, REPLACE WITH YOUR CODE }
    writeln(1, 2);
    flush(output);
    readln(A);

    write('! ')
    for i:=0 to N-1 do
        write(i + 1);
    flush(output);
    
end.

File: OIS/ois_findperm/findperm.cpp
// NOTE: it is recommended to use this even if you don't understand the following code.

#include <fstream>
#include <iostream>

using namespace std;

int main() {
// uncomment the following lines if you want to read/write from files
    // ifstream cin("input.txt");
    // ofstream cout("output.txt");

    int N;
    cin >> N;

    // SAMPLE INTERACTION, REPLACE WITH YOUR CODE

	cout << 1 << ' ' << 2 << '\n';
	cout.flush();
	int x;
	cin >> x;

    cout << "! "; // print the result
    for (int i = 0; i < N; i++)
		cout << i + 1 << ' ';
	cout << endl;

    return 0;
}
